### PR TITLE
add vcpkg

### PR DESCRIPTION
**Pull request by wolfv**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by seanyen** [comment]
Hmm, this one is interesting. What's the usage scenario in your mind? @wolfv :)

**Comment by wolfv** [comment]
Hi @seanyen! My specific usecase is to build some static libraries that I need to link with for `micromamba` (micromamba is a statically linked version of mamba to install conda packages quickly, e.g. in CI or to bootstrap conda environments). Basically I want to _achieve_ the same thing as conda-package-handling which links statically but builds all of the deps in place (https://github.com/conda-forge/conda-package-handling-feedstock/tree/master/recipe) without the complicated recipe (using vcpkg packages for win, linux and os x instead).

We already have a bunch of `-static` libraries on conda-forge but it's a little brittle wrt compiler version changes and there are no win versions for it.

So building the necessary static libraries at the same time as micromamba sounds like a cool idea and vcpkg could simplify this. Also some C++ code in mamba / micromamba is inspired by vcpkg so I think it's really good code as well :)

File: recipes/vcpkg/bld.bat [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+cmake .. ^
+	-G "Ninja" ^
+	-DCMAKE_BUILD_TYPE=Release ^
+	-DVCPKG_DEVELOPMENT_WARNINGS=OFF ^
+	%CMAKE_ARGS%
+
+ninja
+
+MKDIR %LIBRARY_PREFIX%\bin\
+COPY vcpkg.exe %LIBRARY_PREFIX%\bin\
```
Comment (scopatz) [inline recipes/vcpkg/bld.bat:15]: This file needs exit conditions after each command

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**